### DISCUSSION TITLE
Failing to update page

### DISCUSSION THREAD

**Discussion by levilone** [community/community#22232] [Other Feature Feedback, Questions, & Ideas]
<p>I followed the instructions on github pages, the last command didn’t work so I used “git checkout -b master” Then it worked, but now when I try to push updated versions of index.html, no changes occur.</p>

**Comment by airtower-luna** [comment]
<p><code>git checkout -b master</code> would have created a new branch called <code>master</code>, which likely means your already existing code is on an existing branch. The default branch name has been changed to <code>main</code> a while back.</p>
<p>You need to make sure to push to the branch that’s currently published, or if you really want to use a different branch go to the repository setting and select that branch for Pages.</p>

**Comment by levilone** [comment]
<p>I tried git push -u origin main, and got</p>
<blockquote>
<p>Total 0 (delta 0), reused 0 (delta 0)<br>
To <a href="https://github.com/levilone/levilone.github.io" rel="noopener nofollow ugc">https://github.com/levilone/levilone.github.io</a><br>
c75e3d5…dbc355e  main -&gt; main<br>
Branch ‘main’ set up to track remote branch ‘main’ from ‘origin’.</p>
</blockquote>
<p>I tried updating the page with:<br>
git push -u origin main</p>
<p>now when i use git push index.html I get:<br>
fatal: invalid gitfile format: index.html<br>
fatal: Could not read from remote repository.</p>
<p>Please make sure you have the correct access rights<br>
and the repository exists.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="levilone" data-post="3" data-topic="146215">
<div class="title">
<div class="quote-controls"></div>
 levilone:</div>
<blockquote>
<p>now when i use git push index.html I get:</p>
</blockquote>
</aside>
<p>You can’t push a file. You need to commit the file to the branch that’s being published, and then push the branch.</p>
<p>Here’s the Github documentation on pushing:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/github/using-git/pushing-commits-to-a-remote-repository" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="20" height="20">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/github/using-git/pushing-commits-to-a-remote-repository" target="_blank" rel="noopener">Pushing commits to a remote repository - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>For a general introduction to git I highly recommend the <a href="https://git-scm.com/book/en/v2">Git Book</a>, especially chapter 2.</p>

**Comment by levilone** [comment]
<p>thanks for the help, I was able to get it working after some messing around</p>
